driver-core/src/main/com/mongodb/internal/connection/OidcAuthenticator.java

```diff
@@ -185,6 +187,15 @@ public void reauthenticate(final InternalConnection connection) {
         authLock(connection, connection.getDescription());
     }
 
+    @Override
+    public void reauthenticateAsync(final InternalConnection connection, final SingleResultCallback<Void> callback) {
+        assertTrue(connection.opened());
```

Comment:
Let's either add `// method must only be called after original handshake:` here, or better, remove it in `reauthenticate` because the comment does not add any information beyond what the assertion expresses.

Reply:
And rename open to completedHandshake?

Reply:
Will leave same name